### PR TITLE
[WIP] Created daemon restart method

### PR DESCRIPTION
**Pull request by liiight**
### Motivation for changes:
An easy way to reload daemon.

### Detailed changes:

- Created `restart` method
- Renamed `manager,_shutdown` to `_cleanup`
- Made `wait` argument shared between `restart` and `stop` CLI actions
- Added `/server/restart/` API endpoint

Note: This currently only works when running `flexget_vanilla`, more work needed.

### CODE REVIEW

File: flexget/manager.py [hunk 1]
```diff
@@ -916,7 +918,7 @@ def shutdown(self, finish_queue=True):
         fire_event('manager.shutdown_requested', self)
         self.task_queue.shutdown(finish_queue)
 
-    def _shutdown(self):
+    def _cleanup(self):
```
Comment (paranoidi) [inline flexget/manager.py]: This rename is a bit inconvenient as it fires event shutdown ? Seems fine otherwise ...

Comment (liiight) [inline flexget/manager.py]: this only works when running vanilla, making this universal seems a lot more tricky than I thought

**Comment by gazpachoking** [comment]
I think using a native service manager is a better way to handle the daemon start/stop/restart. If there is some way we can make systemd/windows service/other thing easier to set up for users, (without adding too much maintenance burden on us,) I'd prefer that. Not sure if that should be better docs, or included helper scripts or what.

**Comment by paranoidi** [comment]
I would vote to abandon this, being likely way too hard to do well.

**Pull request closed without merging**